model-optimizer/unit_tests/extensions/back/LayoutChangeForEinsum_test.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import unittest
+
+import numpy as np
+
+from extensions.back.LayoutChangeForEinsum import LayoutChangeForEinsum
+from mo.utils.ir_engine.compare_graphs import compare_graphs
+from unit_tests.utils.graph import build_graph
+from mo.front.common.partial_infer.utils import int64_array
+
+
+nodes_attributes = {
+    'placeholder_1': {'shape': None, 'type': 'Parameter', 'kind': 'op', 'op': 'Parameter'},
+    'placeholder_1_data': {'value': None, 'shape': None, 'kind': 'data', 'data_type': None},
+    'placeholder_2': {'shape': None, 'type': 'Parameter', 'kind': 'op', 'op': 'Parameter'},
+    'placeholder_2_data': {'value': None, 'shape': None, 'kind': 'data', 'data_type': None},
+    'placeholder_3': {'shape': None, 'type': 'Parameter', 'kind': 'op', 'op': 'Parameter'},
+    'placeholder_3_data': {'value': None, 'shape': None, 'kind': 'data', 'data_type': None},
+    # Einsum
+    'einsum': {'type': 'Einsum', 'kind': 'op', 'op': 'Einsum'},
+    'einsum_data': {'value': None, 'shape': None, 'kind': 'data'},
+    # Result layer
+    'result': {'type': 'Result', 'kind': 'op', 'op': 'Result'},
+    # Transpose layers
+    'transpose_1': {'type': 'Transpose', 'kind': 'op', 'op': 'Transpose', 'need_shape_inference': True},
+    'transpose_1_data': {'value': None, 'shape': None, 'kind': 'data'},
+    'axis_1_const': {'type': 'Const', 'kind': 'op', 'op': 'Const', 'value': None},
+    'axis_1_const_data': {'kind': 'data', 'value': None, 'shape': None},
+    'transpose_3': {'type': 'Transpose', 'kind': 'op', 'op': 'Transpose', 'need_shape_inference': True},
+    'transpose_3_data': {'value': None, 'shape': None, 'kind': 'data'},
+    'axis_3_const': {'type': 'Const', 'kind': 'op', 'op': 'Const', 'value': None},
+    'axis_3_const_data': {'kind': 'data', 'value': None, 'shape': None},
+}
```

Comment:
Consider using methods `regular_op_with_shaped_data`, `valued_const_with_data` and others like in the model-optimizer/unit_tests/extensions/back/InterpolateReshape_test.py. They allow you to write unit-tests with less code.

Reply:
Fixed.